airflow/api_connexion/parameters.py

```diff
@@ -48,6 +49,22 @@ def format_datetime(value: str):
         )
 
 
+def check_limit(value: int):
+    """
+    This checks the limit passed to view and raises BadRequest if
+    limit exceed user configured value
+    """
+    max_val = conf.getint("api", "maximum_page_limit")  # user configured page limit
```

Comment:
```suggestion
    # user configured page limit
    max_val = conf.getint("api", "maximum_page_limit", fallback= PAGE_LIMIT_DEFAULT)
```

Reply:
Check how I implemented it https://github.com/apache/airflow/pull/9431/commits/49198927cba626957965c962cd098bb018e41df7

Reply:
I tried having the default only on the config but it turned out that limit doesn't get parsed to views. It was giving some unexpected behaviours. But as it is now, it's giving us exactly as described. 
I will also try again if I can have it solely on the config without the one on spec